src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const value of action.keysToRemove) {
+          delete state[value];
+        }
+        break;
+      case 'clear':
+        for (const prop in state) {
+          delete state[prop];
+        }
+        break;
```

Comment:
Let's add a default case to avoid potential errors if some type doesn't match any of cases
![image](https://user-images.githubusercontent.com/96409563/208066388-b9bb9239-43f3-44d1-9d87-cd3dc7a3acd8.png)